tests/erlang_tests/bigint.erl

```diff
@@ -0,0 +1,107 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2025 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(bigint).
+-export([start/0, mul/2, shrink/0, pow/2, twice/1, fact/1, get_machine_atom/0, expect_overflow/1]).
+
+start() ->
+    test_mul().
+
+test_mul() ->
+    Expected_INT64_MIN = ?MODULE:pow(-2, 63),
+    Expected_INT64_MIN = ?MODULE:shrink(),
+    A = ?MODULE:mul(16#10101010CAFECAFE, 16#AABB),
+    Square = ?MODULE:mul(A, A),
+    <<"2559181265480533323615999200984578944503596644">> = erlang:integer_to_binary(Square),
```

Comment:
```
    2559181265480533323615999200984578944503596644 = Square,
```

We also need support for literal decoding.

Reply:
Indeed, we are getting there step by step.